nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/AddPermissionTest.java

```diff
@@ -421,6 +426,37 @@ public void shouldAddPermissionForUserWhenBlockingInheritance() throws Operation
         assertExpectedPermissions(acl, params, "existingUser1");
     }
 
+    @Test
+    public void shouldAddPermissionForUsersWhenUsingChainExtension() throws OperationException {
+        // Mock the users params
+        List<String> users = Arrays.asList("jchastain", "rdeniro", "dwashington");
+        users.forEach(user -> when(userManager.getUserModel(user)).thenReturn(new SimpleDocumentModel(user)));
+
+        DocumentModel doc = session.getDocument(new PathRef("/src"));
+        OperationContext ctx = new OperationContext(session);
+        ctx.setInput(doc);
+
+        Object object = automationService.run(ctx, "testAddPermissionToUsers");
+        assertThat(object, instanceOf(DocumentModel.class));
+
+        DocumentModel documentModel = (DocumentModel) object;
+
+        ACL acl = documentModel.getACP().getACL("local");
+        assertNotNull("Local ACL cannot be null !", acl);
+        assertThat("ACEs size must be the same then users size !", acl.getACEs(), arrayWithSize(users.size()));
```

Comment:
I would remove and simplify the whole check of permissions here by using the helper `assertExpectedPermissions` method, filling the `params` Map with the expected values.

```java
assertExpectedPermissions(acl, params, "jchastain");
assertExpectedPermissions(acl, params, "rdeniro");
...
```